### ISSUE TITLE
Delete should be executed before adding new files

### ISSUE THREAD

**Issue by PhyxionNL** [Jumoo/uSync.Complete.Issues#241]
**Describe the bug**
Say you have a folder with files named 1, 2, and 3. Remove these, and upload three new ones also named 1, 2, and 3. Push this folder to destination, what uSync does is first create these files so they're now named 1 (1), 2 (1) and 3 (1). Then the original 1, 2, 3 are removed.
You now need to push twice to fix the names.

**Version (please complete the following information):**
 - Umbraco Version: 13.2.2
 - uSync.Complete Version: 13.1.5

**To Reproduce**
1. See description.

**Expected behavior**
Execute delete before adding new items.

**Comment by KevinJump** [comment]
yeah we will need to look, we have the opposite issues also (delete should happen last). 

if the deletes happen before the updates then content that might be in the folders would be deleted, (so it needs to be moved out of the folders via its own saves before the deletes then happen).

We do have some dependency graph stuff in there but not for deletes before saves, because we are not running it as a full event audit log (e.g we only know what we have been sent, at push time, not all the actions that might have happened in between). its hard for us to say, oh this was all moved **then** this was deleted.

TLDL: We will look to see what we can do.

**Comment by PhyxionNL** [comment]
There are probably cases where a delete should happen last, but I don't think that's logical here.
If I sync content from X to Y, I would expect that Y becomes identical to X. If a node already exists with the same name, it should be removed as otherwise you'll get some mishmash. Renaming to ... (1) is probably not something someone ever wants, especially for _newly added_ content.
